src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs

```diff
@@ -11,6 +11,7 @@ namespace System.Net.Security
 {
     internal sealed class SslAuthenticationOptions
     {
+        private const string EnableOcspStaplingContextSwitchName = "System.Net.Security.EnableOcspStapling";
```

Comment:
IT Looks good enough to me. But would do negative  since it is enabled by default. e.g. `System.Net.Security.DisableOcspStapling` If we want to be verbose we can do `DisableTlsOcspStapling` but I think it really make sense only in context of SSL/TLS.

Reply:
My point was that the switch does not turn if off globally, but only when certain cert selection methods were used for SslStream, so the name might be misleading, but since I don't have a better idea then I am fine with the way it is.

Reply:
Isn't it disabled by default? Since, we're passing `true` to `noOcspFetch` parameter?

Reply:
> My point was that the switch does not turn if off globally, but only when certain cert selection methods were

can we make it so? At the moment it impacts only server side on Linux, right? What I was suggesting is to make it impact client as well. And preferably Winoews is well if there is something we can do via API

Reply:
> can we make it [turn off OCSP Stapling globally]?

That's the opposite of what the original issue is asking for, though.

It used to be the case that starting SslStream with just a certificate turned on OCSP Stapling, because it created the cert context to unify the code path.  Then, for whatever reason (presumably it was a good one, but I didn't look it up right now and don't remember it) there was a change to distinguish cert contexts created by the user (stapling on) and contexts created implicitly by the user only offering a certificate.  In the issue someone wants OCSP Stapling to be turned back on in the "from cert" case.

So, while making a "disable OCSP stapling" app context switch is sort of doable (I don't know how to stop macOS from doing it.. assuming it does it...), it's not something that would address the requestor's problem.